### PR TITLE
unix: fix uv_uptime() on linux

### PR DESCRIPTION
**Pull request by schamberg97**
This change fixes https://github.com/libuv/libuv/issues/3068 on OpenVZ Containers in Linux.

### CODE REVIEW

**Comment by schamberg97** [comment]
Running `make check` on OpenVZ system with the patch produces this:

```
ok 1 - platform_output
# Output from process `platform_output`:
# uv_get_process_title: /home/nik/libuv/test/.libs/run-tests
# uv_cwd: /home/nik/libuv
# uv_resident_set_memory: 1056768
# uv_uptime: 621523.990000
...
...
```

checking `/proc/uptime/` a bit later:

```
nik@vps1607355603:~/libuv$ cat /proc/uptime
621534.08 621342.55
```

Just to check that Host uptime is significantly bigger (with `clock_gettime`) and I have indeed solved the issue:

Code used: 

```
#include <time.h>
#include <errno.h>
#include <stdio.h>

int main() {
  struct timespec now;
  int r;

  r = clock_gettime(CLOCK_BOOTTIME, &now);
  if (r != 0) {
    fprintf(stderr, "clock_gettime failed with %d\n", errno);
    return errno;
  }
  fprintf(stderr, "system uptime: %ld\n", now.tv_sec);
  return 0;
}
```

result after compiling and running: `system uptime: 1899806`

File: src/unix/linux-core.c [hunk 1]
```diff
@@ -602,29 +602,63 @@ int uv_resident_set_memory(size_t* rss) {
...
+  return 0;
+}
 
 int uv_uptime(double* uptime) {
   static volatile int no_clock_boottime;
+  char buf[128];
   struct timespec now;
   int r;
 
+  // Try /proc/uptime first, then fallback to the previous implementation
```
Comment (bnoordhuis) [inline src/unix/linux-core.c:636]:
```suggestion
  /* Try /proc/uptime first, then fallback to clock_gettime(). */
```
(C89 comments + it won't be clear to a casual reader what "previous implementation" refers to.)

File: src/unix/linux-core.c [hunk 2]
```diff
@@ -602,29 +602,63 @@ int uv_resident_set_memory(size_t* rss) {
...
+  
+  if (0 == uv__slurp("/proc/uptime", buf, sizeof(buf)) ) {
+    if (1 == sscanf(buf, "%lf", uptime) ) {
+      return 0;
+    }
+  }
```
Comment (bnoordhuis) [inline src/unix/linux-core.c:642]:
```suggestion
  if (0 == uv__slurp("/proc/uptime", buf, sizeof(buf)))
    if (1 == sscanf(buf, "%f", uptime))
      return 0;
```
(Style + `%lf` is for `long double`.)

File: src/unix/linux-core.c [hunk 3]
```diff
@@ -602,29 +602,63 @@ int uv_resident_set_memory(size_t* rss) {
...
-    goto retry;
+    goto retry_clock_gettime;
   }
 
   if (r)
     return UV__ERR(errno);
 
   *uptime = now.tv_sec;
   return 0;
+
```
Comment (bnoordhuis) [inline src/unix/linux-core.c:661]:
```suggestion
```

File: src/unix/linux-core.c [hunk 4]
```diff
@@ -1024,33 +1058,6 @@ void uv__set_process_title(const char* title) {
 #endif
 }
 
-
```
Comment (bnoordhuis) [inline src/unix/linux-core.c:1027]: Two blank lines here, please.

**Comment by schamberg97** [comment]
@bnoordhuis I've made the requested changes

**Review by bnoordhuis — APPROVED** [review]
LGTM, thanks!

**Review by santigimeno — APPROVED** [review]
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/2102/

**Comment by santigimeno** [comment]
Landed in https://github.com/libuv/libuv/commit/77a2394fed7ddca98ed7b72df73cd8469867ade1. Thanks!

**Pull request closed without merging**